### PR TITLE
Use new tfplugins docs

### PR DESCRIPTION
**Pull request by shinji62**
### Description
Updated tfdocs plugins
Updated repository structure to support the new tfdocs plugins 
Fixed some typos

### CODE REVIEW

File: templates/resources/universe.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+1. Toggle TLS settings
+1. Editing cluster parameters
+    1. Instance type
+    1. Number of Nodes
+    1. Number of Volumes per instance
+    1. Volume Size
+    1. User Tags
+1. Delete read replicas (Adding read replica after universe creation currently not supported)
+
+{{ .SchemaMarkdown | trimspace }}
```
Comment (Deepti-yb) [inline templates/resources/universe.md.tmpl:35]: Import statement for yba_universe has to be added here.

File: go.mod [hunk 2]
```diff
@@ -6,67 +6,55 @@ require (
 	github.com/aws/aws-sdk-go v1.44.122
 	github.com/bramvdbogaerde/go-scp v1.2.0
 	github.com/google/uuid v1.3.0
-	github.com/hashicorp/terraform-plugin-docs v0.5.1
+	github.com/hashicorp/terraform-plugin-docs v0.16.0
 	github.com/hashicorp/terraform-plugin-log v0.7.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.10.1
 	github.com/yugabyte/platform-go-client v0.0.0-20230810122221-e283fa78cb44
-	golang.org/x/crypto v0.0.0-20220315160706-3147a52a75dd
+	golang.org/x/crypto v0.8.0
```
Comment (Deepti-yb) [inline go.mod:13]: Do we have to update all the imported modules to the latest version in this PR? Would it work just to update the terraform-plugin-docs version for now?

If the other updates are required, does it break any of the existing workflows?

File: templates/data-sources/backup_info.md.tmpl
Comment (Deepti-yb) [inline templates/data-sources/backup_info.md.tmpl:1]: Can we add a new line at the end of each file in the template directory to maintain markdown file syntax?

**Review by Deepti-yb — CHANGES_REQUESTED** [review]
Thank you for this major update, just added a few comments.

Since we are updating module dependencies, we need to include the correct licenses in the NOTICE file. This can be done by running `./licenses/generate-license-file.sh` in the root folder of this project.

File: templates/resources/universe.md.tmpl (hunk 1 above)
Comment (shinji62) [reply templates/resources/universe.md.tmpl:35]: Oh yeah will do thanks.

File: templates/data-sources/backup_info.md.tmpl
Comment (shinji62) [reply templates/data-sources/backup_info.md.tmpl:1]: I can do that as well.

File: go.mod (hunk 2 above)
Comment (shinji62) [reply go.mod:13]: I did update but I didn't see any issue during the test .

File: templates/resources/universe.md.tmpl (hunk 1 above)
Comment (shinji62) [reply templates/resources/universe.md.tmpl]: done, please check and resolve comment if ok

File: templates/data-sources/backup_info.md.tmpl
Comment (shinji62) [reply templates/data-sources/backup_info.md.tmpl:1]: done, please check and resolve comment if ok

File: go.mod (hunk 2 above)
Comment (shinji62) [reply go.mod:13]: done, please check and resolve comment if ok

**Review by Deepti-yb — COMMENTED** [review]
Thank you for the changes. I'll just run a quick check on the workflows.

**Review by Deepti-yb — APPROVED** [review]

**Pull request merged**